### PR TITLE
feat(arch): move && simplify component select

### PR DESCRIPTION
**Pull request by dpellier**
⚠ BREAKING CHANGES ⚠
- OsdsSelect is no longer part of `@ovhcloud/ods-stencil` package
- OsdsSelect is now accessible by adding either `ods-component-select` or `ods-components` dependencies
- `@ovhcloud/ods-stencil-select` lib has been replaced by `@ovhcloud/ods-component-select`
- `@ovhcloud/ods-stencil-select-react` lib has been replaced by `@ovhcloud/ods-component-select-react`
- `@ovhcloud/ods-stencil-select-vue` lib has been replaced by `@ovhcloud/ods-component-select- vue`
- `OdsSelectSize` has been removed from `@ovhcloud/ods-core` and is replaced by `ODS_SELECT_SIZE` from `@ovhcloud/ods-component-select`
- `OdsSelectSizeList` has been removed from `@ovhcloud/ods-core` and is replaced by `ODS_SELECT_SIZES` from `@ovhcloud/ods-component-select`
- `OdsSelectSizeUnion` has been removed from `@ovhcloud/ods-core`
- OsdsSelect is now displayed as flex by default
- OsdsSelect `flex` attribute has been removed
- OsdsSelect `inline` attribute has been added
- `@ovhcloud/ods-component-select` exports the following interfaces: `OdsSelectAttribute`, `OdsSelectEvent`, `OdsSelectValueChangeEvent`, `OdsSelectValueChangeEventDetail`, `OdsSelectMethod`, `OdsSelectOptionAttribute`, `OdsSelectOptionEvent`, `OdsSelectOptionClickEventDetail`and `OdsSelectOptionMethod`

### CODE REVIEW

File: packages-new/components/select/src/components/osds-select-group/osds-select-group.tsx [hunk 1]
```diff
@@ -18,8 +11,7 @@ import { OdsStencilEvents, OdsStencilMethods } from '@ovhcloud/ods-stencil/libra
   styleUrl: 'osds-select-group.scss',
   shadow: true,
 })
-export class OsdsSelectGroup implements OdsSelectGroup<OdsStencilMethods<OdsSelectGroupMethods>, OdsStencilEvents<OdsSelectGroupEvents>> {
-  controller: OdsSelectGroupController = new OdsSelectGroupController(this);
+export class OsdsSelectGroup {
```
Comment (aesteves60) [inline packages-new/components/select/src/components/osds-select-group/osds-select-group.tsx:14]: export end of file

File: packages-new/components/select/src/components/osds-select-option/osds-select-option.spec.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  describe('contents', () => {
+    it('should have a placeholder slot', async () => {
+      await setup({ attributes: {} });
+      expect(slotPlaceholder).toBeTruthy();
+    });
+  });
+
+  describe('attributes', () => {
+
+    // function getAttributeContextOptions<Name extends keyof OdsSelectOptionAttributes = keyof OdsSelectOptionAttributes>({
```
Comment (aesteves60) [inline packages-new/components/select/src/components/osds-select-option/osds-select-option.spec.ts:55]: clean the comment

File: packages-new/components/select/src/components/osds-select-option/osds-select-option.tsx [hunk 3]
```diff
@@ -22,10 +17,9 @@ import { OdsStencilEvents, OdsStencilMethods } from '@ovhcloud/ods-stencil/libra
   styleUrl: 'osds-select-option.scss',
   shadow: true,
 })
-export class OsdsSelectOption implements OdsSelectOption<OdsStencilMethods<OdsSelectOptionMethods>, OdsStencilEvents<OdsSelectOptionEvents>> {
+export class OsdsSelectOption implements OdsSelectOptionAttribute, OdsSelectOptionEvent, OdsSelectOptionMethod {
```
Comment (aesteves60) [inline packages-new/components/select/src/components/osds-select-option/osds-select-option.tsx:20]: export eof

File: packages-new/components/select/src/components/osds-select/core/controller.spec.ts [hunk 4]
```diff
@@ -1,32 +1,51 @@
...
   beforeEach(() => {
-    component = new OdsSelectMock();
-    item1 = document.createElement('osds-select-option') as OdsSelectOption & HTMLElement;
-    item1.value = '1';
-    item2 = document.createElement('osds-select-option') as OdsSelectOption & HTMLElement;
-    item2.value = '2';
-    component.el = document.createElement('osds-select');
-    component.el.appendChild(item1);
-    component.el.appendChild(item2);
+    // component = new OdsSelectMock();
```
Comment (aesteves60) [inline packages-new/components/select/src/components/osds-select/core/controller.spec.ts:41]: remove comment

File: packages-new/components/select/src/components/osds-select/core/controller.ts [hunk 5]
```diff
@@ -78,6 +78,7 @@ export class OdsSelectController extends OdsComponentController<OdsSelect> {
   }
 
   handlerKeyDown(event: KeyboardEvent): void {
+    console.log(this.selectOptions)
```
Comment (aesteves60) [inline packages-new/components/select/src/components/osds-select/core/controller.ts:81]: oups

File: packages-new/components/select/src/components/osds-select/interfaces/attributes.ts [hunk 6]
```diff
@@ -30,17 +29,21 @@ export interface OdsSelectAttributes extends OdsComponentAttributes {
...
+  size: ODS_SELECT_SIZE;
   /*
    * color of the select input and his icon arrow.
    * color is not yet customizable trough his attribute.
    * for now, it is forced to primary color (internally)
    * @experimental
    */
   /** the primary color of the theme */
-  color: OdsThemeColorIntent.primary;
+  color: OdsThemeColorIntent;
```
Comment (aesteves60) [inline packages-new/components/select/src/components/osds-select/interfaces/attributes.ts:42]: alphabet order

File: packages-new/components/select/src/components/osds-select/interfaces/methods.ts [hunk 7]
```diff
@@ -1,36 +1,39 @@
...
    * set tab index on the component
    */
-  setInputTabindex(value: number): void;
+  setInputTabindex(value: number): Promise<void>;
 
   /**
    * get the validity state
    */
-  getValidity(): OdsSelectValidityState;
+  getValidity(): Promise<OdsValidityState>;
```
Comment (aesteves60) [inline packages-new/components/select/src/components/osds-select/interfaces/methods.ts:34]: alphabet order

File: packages-new/components/select/src/components/osds-select/osds-select.spec.ts [hunk 8]
```diff
@@ -76,87 +67,97 @@ describe('spec:osds-select', () => {
...
-        root: () => page.root,
-        wait: () => page.waitForChanges(),
-        setup: (value) => {
-          const attributes: Partial<OdsSelectAttributes> = {};
-          attributes[ name ] = value;
-          return setup({ attributes });
-        },
-      })
-    }
+    // function getAttributeContextOptions<Name extends keyof OdsSelectAttributes = keyof OdsSelectAttributes>({
```
Comment (aesteves60) [inline packages-new/components/select/src/components/osds-select/osds-select.spec.ts:70]: clean comment

File: packages-new/components/select/src/components/osds-select/osds-select.tsx [hunk 9]
```diff
@@ -31,7 +27,7 @@ ocdkDefineCustomElements()
   styleUrl: 'osds-select.scss',
   shadow: true,
 })
-export class OsdsSelect implements OdsSelect<OdsStencilMethods<OdsSelectMethods>, OdsStencilEvents<OdsSelectEvents>> {
+export class OsdsSelect implements OdsSelectAttribute, OdsSelectEvent, OdsSelectMethod {
```
Comment (aesteves60) [inline packages-new/components/select/src/components/osds-select/osds-select.tsx:30]: export eof

File: packages/stencil/components/stencil.config.ts [hunk 10]
```diff
@@ -42,6 +42,7 @@ const config: Config = {
         'osds-icon',
         'osds-input',
         'osds-link',
+        'osds-select',
```
Comment (aesteves60) [inline packages/stencil/components/stencil.config.ts:45]: should we add `osds-select-option` & `osds-select-group` ?

**Review by aesteves60 — CHANGES_REQUESTED** [review]

File: packages-new/components/select/src/components/osds-select-group/osds-select-group.tsx (hunk 1 above)
Comment (aesteves60) [reply packages-new/components/select/src/components/osds-select-group/osds-select-group.tsx:14]: I think that's don't build if the export is at the end

File: packages-new/components/select/src/components/osds-select-option/osds-select-option.tsx (hunk 3 above)
Comment (aesteves60) [reply packages-new/components/select/src/components/osds-select-option/osds-select-option.tsx:20]: I think that's don't build if the export is at the end

File: packages-new/components/select/src/components/osds-select/osds-select.tsx (hunk 9 above)
Comment (aesteves60) [reply packages-new/components/select/src/components/osds-select/osds-select.tsx:30]: I think that's don't build if the export is at the end

File: packages/stencil/components/stencil.config.ts (hunk 10 above)
Comment (dpellier) [reply packages/stencil/components/stencil.config.ts:45]: I don't think so

**Pull request merged**